src/tools/perf/perftest_run.c

```diff
@@ -262,10 +262,28 @@ static ucs_status_t read_batch_file(FILE *batch_file, const char *file_name,
     return UCS_OK;
 }
 
+static size_t get_test_count(FILE *batch_file)
+{
+    size_t test_count = 0;
+
+#define MAX_ARG_SIZE 2048
+    char buf[MAX_ARG_SIZE];
+
+    while (fgets(buf, sizeof(buf) - 1, batch_file) != NULL) {
+        if ((buf[0] != '#') && (strtok(buf, " \t\n\r"))) {
```

Comment:
what is the format of batch_file? with lines containing only one space, we would count them as test? same with leading spaces before `#`?

Reply:
Yes, you are right, now I see
Ideally we want to reuse the same code for parsing, let me try to implement that

Reply:
done